runtime/src/lib.rs

```diff
@@ -513,6 +513,39 @@ impl pallet_vesting::Config for Runtime {
 	const MAX_VESTING_SCHEDULES: u32 = 28;
 }
 
+pub struct RelayChainBlockNumberProvider<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: cumulus_pallet_parachain_system::Config> BlockNumberProvider
+for RelayChainBlockNumberProvider<T>
+{
+	type BlockNumber = BlockNumber;
+
+	fn current_block_number() -> Self::BlockNumber {
+		cumulus_pallet_parachain_system::Pallet::<T>::validation_data()
+			.map(|d| d.relay_parent_number)
+			.unwrap_or_default()
+	}
+}
+
+parameter_types! {
+
+	pub FutureAuctionTreasuryId: AccountId = PalletId(*b"Tresury1").into_account();
```

Comment:
We should discuss and make decisions about naming this pots

Reply:
What I saw how other pallets are working is to here provide only PalletId, and then inside the pallet to convert it into account when is needed